pysma/__init__.py

```diff
@@ -255,16 +255,18 @@ def read(self, sensors):
                 return False
         body = yield from self._fetch_json(URL_VALUES, payload=payload)
 
-        # On the first 401 error we close the session which will re-login
-        if body.get("err") == 401:
+        # On the first error we close the session which will re-login
+        err = body.get("err")
+        if err is not None:
             _LOGGER.warning(
-                "401 error detected, closing session to force " "another login attempt"
+                "%s: error detected, closing session to force another login attempt, got: %s", self._url, body
             )
-            self.close_session()
+            yield from self.close_session()
```

Comment:
This change appears to be the essential one. Without `yield from` here, the session never closed.